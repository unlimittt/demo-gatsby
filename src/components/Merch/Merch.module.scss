@import "../../assets/scss/variables";
@import "../../assets/scss/mixins";

.merch {
  &__title {
    font-size: $typographic-base-font-size * 2.5;
    font-weight: 600;
    @include line-height(2);
    @include margin-top(0);
    @include margin-bottom(1.45);
  }

  &__body {
    & figure {
      @include margin-bottom(1);

      & blockquote {
        font-style: italic;
        text-align: center;
        margin-top: 0;
        @include padding(1, 0);

        & p {
          max-width: $layout-post-width;
          font-size: $typographic-base-font-size * 1.6817;
          margin-top: 0;
          @include margin-bottom(1);
          @include line-height(1.5);
        }
      }
    }

    & a {
      text-decoration: underline;
    }

    & * {
      max-width: $layout-post-width;
      margin-left: auto;
      margin-right: auto;
    }

    & img {
      max-width: 100%;
    }
  }

  &__price {
    text-align: center;
    font-size: $typographic-base-font-size;
    font-weight: 600;
    @include line-height(1.5);
  }

  &__buy-button {
    float: right;
    display: inline-block;
    height: $button-height;
    padding: 0 24px;
    line-height: $button-height;
    border: 1px solid $color-gray-border;
    text-decoration: none;
    border-radius: $button-border-radius;
    color: $color-base;

    &:hover,
    &:focus {
      color: $color-primary;
    }
  }
}

@include breakpoint-sm {
  .page {
    lost-column: 7/12;

    &__inner {
      padding: 30px 20px;
    }
  }
}

@include breakpoint-md {
  .page {
    lost-column: 2/3;

    &__inner {
      padding: 40px 35px;
    }
  }
}
