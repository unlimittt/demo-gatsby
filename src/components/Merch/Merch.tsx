import React from "react"
import styles from "./Merch.module.scss"
import { MarkdownRemark } from "../../types"
import Img from "gatsby-image"
import { Link } from "gatsby"

type Props = {
  merch: MarkdownRemark
}

const Merch = ({ merch }: Props) => (
  <div className={styles["merch"]}>
    <h1 className={styles["merch__title"]}>{merch.frontmatter.title}</h1>
    <Img fluid={merch.frontmatter.image.childImageSharp.fluid} />
    <div className={styles["merch__price"]}>
      Price: $ {merch.frontmatter.price}
    </div>
    <div className={styles["merch__body"]} dangerouslySetInnerHTML={{ __html: merch.html }} />
    <Link to="#"
      className={`snipcart-add-item  ${styles["merch__buy-button"]}`}
      data-item-id={merch.fields.slug}
      data-item-price={merch.frontmatter.price}
      data-item-url={merch.fields.slug}
      data-item-image={merch.frontmatter.image.childImageSharp.fluid.src}
      data-item-name={merch.frontmatter.title}
    >
      BUY NOW
    </Link>
  </div>
)

export default Merch
