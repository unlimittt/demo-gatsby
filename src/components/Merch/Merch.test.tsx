import React from "react"
import Enzyme, { shallow } from "enzyme"
import Adapter from "enzyme-adapter-react-16"
import Merch from "./Merch"

Enzyme.configure({ adapter: new Adapter() })

describe("Merch", () => {
  const props = {
    merch: {
      fields: {
        slug: "/merch/personalized-bottle",
      },
      frontmatter: {
        title: "Personalized Bottle",
        image: {
          childImageSharp: {
            fluid: {
              aspectRatio: 1,
              src: "/media/customized-coke-bottle.png",
              srcSet: "",
              sizes: "",
            },
          },
        },
        order: 1,
        price: "20.00",
      },
      html: "<p>test</p>",
      id: "test_1",
    },
  }

  it("renders correctly", () => {
    const wrapper = shallow(<Merch {...props} />)
    expect(wrapper).toMatchSnapshot()
  })
})
