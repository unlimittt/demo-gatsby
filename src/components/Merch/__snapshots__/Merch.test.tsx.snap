// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Merch renders correctly 1`] = `
<div
  className="merch"
>
  <h1
    className="merch__title"
  >
    Personalized Bottle
  </h1>
  <Image
    Tag="div"
    alt=""
    durationFadeIn={500}
    fadeIn={true}
    fluid={
      Object {
        "aspectRatio": 1,
        "sizes": "",
        "src": "/media/customized-coke-bottle.png",
        "srcSet": "",
      }
    }
    loading="lazy"
  />
  <div
    className="merch__price"
  >
    Price: $ 
    20.00
  </div>
  <div
    className="merch__body"
    dangerouslySetInnerHTML={
      Object {
        "__html": "<p>test</p>",
      }
    }
  />
  <mockConstructor
    className="snipcart-add-item  merch__buy-button"
    data-item-id="/merch/personalized-bottle"
    data-item-image="/media/customized-coke-bottle.png"
    data-item-name="Personalized Bottle"
    data-item-price="20.00"
    data-item-url="/merch/personalized-bottle"
    to="#"
  >
    BUY NOW
  </mockConstructor>
</div>
`;
